R-package/R/lgb.Booster.R

```diff
@@ -39,12 +39,12 @@ Booster <- R6::R6Class(
         # Check if training dataset is not null
         if (!is.null(train_set)) {
           # Check if training dataset is lgb.Dataset or not
-          if (!lgb.check.r6.class(train_set, "lgb.Dataset")) {
+          if (!(lgb.Dataset = lgb.check.r6.class(train_set, "lgb.Dataset"))) {
```

Comment:
this code will result in overwriting the function `lgb.Dataset` with a boolean value. Please revert it by accepting this suggestion in the browser.
```suggestion
          if (!lgb.check.r6.class(object = train_set, name = "lgb.Dataset")) {
```